src/infrasys/in_memory_time_series_storage.py

```diff
@@ -22,18 +25,32 @@ class InMemoryTimeSeriesStorage(TimeSeriesStorageBase):
     """Stores time series in memory."""
 
     def __init__(self) -> None:
-        self._arrays: dict[UUID, TimeSeriesData] = {}  # Time series UUID, not metadata UUID
+        self._arrays: dict[UUID, NDArray] = {}  # Time series UUID, not metadata UUID
 
     def get_time_series_directory(self) -> None:
         return None
 
+    def iter_time_series_uuids(self) -> Iterable[UUID]:
+        return self._arrays.keys()
+
     def add_time_series(self, metadata: TimeSeriesMetadata, time_series: TimeSeriesData) -> None:
         if metadata.time_series_uuid not in self._arrays:
-            self._arrays[metadata.time_series_uuid] = time_series
+            time_series_array = time_series.data
```

Comment:
will update to time_series.data_array

Reply:
@daniel-thom, this is where I needed to reference .data. There is no other way to access the data from TimeSeriesData without enforcing data on the base class. This method could be changed to take SingleTimeSeries as an input instead of TimeSeriesData

Reply:
But I really think it makes sense to add data as an attribute to the base class, I was messing with generics so that we could create SingleTimeSeries[pint.Quantity] or something like that to type the data attribute in any child class. I don't think I understand the case where a child class of TimeSeriesData wouldn't have a data attribute

Reply:
You could do something similar to what is done in `get_time_series` and throw an error if the type is not SingleTimeSeries.

I agree that all subclasses of TimeSeriesData will have some `data` attribute. But they could look very different and have other supporting attributes. They could have a different number of dimensions. It could look like this [Sienna type](https://github.com/NREL-Sienna/InfrastructureSystems.jl/blob/main/src/deterministic.jl#L25). The exact type should be in the subclass. The main point is that any method that operates on data should know what the time series type is. I'm not sure that any code can do anything meaningful with `TimeSeriesData.data`. Do we want to do something that requires subclasses to have the exact field name `data` for debugging. I suppose there is some value in that.